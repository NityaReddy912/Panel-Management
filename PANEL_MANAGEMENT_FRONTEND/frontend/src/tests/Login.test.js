import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { screen, configure } from '@testing-library/react';
import LoginPage from "../Components/AddList/AddForm";
import { BrowserRouter as Router } from 'react-router-dom';
import {MemoryRouter} from 'react-router-dom';
jest.mock('../Components/Login/Login');

describe("Add Component", () => {
    function addPanel(){
      return render( 
      <MemoryRouter intialEntries={['/login']}>
        <LoginPage />
      </MemoryRouter>
      )
    }
    it('login', ()=>{
      screen.debug();
      expect(addPanel()).toMatchSnapshot();
    });

    it('Password', () =>{
      const {getByRole} = addPanel();
      const element = screen.queryByTestId('login-password');
    });

    it('Login', () =>{
      const {getByRole} = addPanel();
      const element = screen.queryByTestId('login-submit-btn');
    });
})
