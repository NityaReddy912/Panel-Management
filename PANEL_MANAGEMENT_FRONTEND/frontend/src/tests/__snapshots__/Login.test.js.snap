// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Component login 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-wrap"
      >
        <h2
          class="addHead"
        >
          Add 
           Master
        </h2>
        <form
          class="addList__form"
        >
          <div
            class="row"
          >
            <div
              class="col-md-1"
            >
              <div
                class="form-group"
              >
                <label
                  class="fw-bolder"
                  for="first"
                >
                   Name
                </label>
              </div>
            </div>
            <div
              class="col-md-5"
            >
              <div
                class="form-group"
              >
                <input
                  class="form-control  false "
                  id="roleName"
                  placeholder="Enter undefined Name"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="row btnGroup addList__button"
          >
            <div
              class="col-md-2"
            >
              <button
                class="btn btn-primary"
                type="submit"
              >
                Submit
              </button>
            </div>
            <div
              class="col-md-3"
            >
              <button
                class="btn btn-primary btnCancel"
              >
                Cancel
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-wrap"
    >
      <h2
        class="addHead"
      >
        Add 
         Master
      </h2>
      <form
        class="addList__form"
      >
        <div
          class="row"
        >
          <div
            class="col-md-1"
          >
            <div
              class="form-group"
            >
              <label
                class="fw-bolder"
                for="first"
              >
                 Name
              </label>
            </div>
          </div>
          <div
            class="col-md-5"
          >
            <div
              class="form-group"
            >
              <input
                class="form-control  false "
                id="roleName"
                placeholder="Enter undefined Name"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="row btnGroup addList__button"
        >
          <div
            class="col-md-2"
          >
            <button
              class="btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </div>
          <div
            class="col-md-3"
          >
            <button
              class="btn btn-primary btnCancel"
            >
              Cancel
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
